k8s/upgrade/src/bin/upgrade-job/common/error.rs

```diff
@@ -214,6 +214,11 @@ pub(crate) enum Error {
         namespace: String,
     },
 
+    /// Error for when a Kubernetes API request for GET-ing a list of Nodes filtered by label(s)
+    /// fails.
+    #[snafu(display("Failed to list Nodes with label {}: {}", label, source))]
```

Comment:
should source be part of display? This will cause duplicates when dumping the error list

Reply:
If they're only logged as such then it's fine I guess, the problem arises when you want to inspect the entire error chain as you start getting duplicate messages.
The "difficulty" you're referring to is because you're only relying in the proc display implementation.
Seems the current way in this file is already doing it this way so its fine for now.